pkg/block/s3/adapter.go

```diff
@@ -507,36 +434,43 @@ func (a *Adapter) GetProperties(ctx context.Context, obj block.ObjectPointer) (b
 		Bucket: aws.String(bucket),
 		Key:    aws.String(key),
 	}
-	client := a.clients.Get(ctx, qualifiedKey.GetStorageNamespace())
-	s3Props, err := client.HeadObjectWithContext(ctx, headObjectParams)
+	client := a.clients.Get(ctx, bucket)
+	s3Props, err := client.HeadObject(ctx, headObjectParams)
 	if err != nil {
 		return block.Properties{}, err
 	}
-	return block.Properties{StorageClass: s3Props.StorageClass}, nil
+	return block.Properties{
+		StorageClass: aws.String(string(s3Props.StorageClass)),
+	}, nil
 }
 
 func (a *Adapter) Remove(ctx context.Context, obj block.ObjectPointer) error {
 	var err error
 	defer reportMetrics("Remove", time.Now(), nil, &err)
-	bucket, key, qualifiedKey, err := a.extractParamsFromObj(obj)
+	bucket, key, _, err := a.extractParamsFromObj(obj)
 	if err != nil {
 		return err
 	}
-	deleteObjectParams := &s3.DeleteObjectInput{
+
+	deleteInput := &s3.DeleteObjectInput{
 		Bucket: aws.String(bucket),
 		Key:    aws.String(key),
 	}
-	svc := a.clients.Get(ctx, qualifiedKey.GetStorageNamespace())
-	_, err = svc.DeleteObjectWithContext(ctx, deleteObjectParams)
+	client := a.clients.Get(ctx, bucket)
+	_, err = client.DeleteObject(ctx, deleteInput)
 	if err != nil {
 		a.log(ctx).WithError(err).Error("failed to delete S3 object")
 		return err
 	}
-	err = svc.WaitUntilObjectNotExistsWithContext(ctx, &s3.HeadObjectInput{
+
+	// TODO(barak): check if wait for delete is needed here
```

Comment:
TODO?

Reply:
Deleted the TODO - kept the current implementation as the only client is a test.
We should consider removing the wait if we plan to use this call in non-test.